tests/cnc_test.c

```diff
@@ -37,6 +43,35 @@ void test_missing_config(void)
 	CU_ASSERT_EQUAL(result, -1);
 }
 
+void test_backup_full(void)
+{
+	ini_config = (config_t *)malloc(sizeof(config_t));
+	ini_config->postgres_config = (postgres_t *)malloc(sizeof(postgres_t));
+	ini_config->smtp_config = (smtp_t *)malloc(sizeof(smtp_t));
+	ini_parse_string("[postgres]\nbackupt_type=full", handler, ini_config);
+	CU_ASSERT_EQUAL(ini_config->postgres_config->backup_type, FULL);
+}
+
+void test_backup_schema_only(void)
+{
+	ini_config = (config_t *)malloc(sizeof(config_t));
+	ini_config->postgres_config = (postgres_t *)malloc(sizeof(postgres_t));
+	ini_config->smtp_config = (smtp_t *)malloc(sizeof(smtp_t));
+	ini_parse_string("[postgres]\nbackup_type=schema", handler, ini_config);
+	CU_ASSERT_EQUAL(ini_config->postgres_config->backup_type, SCHEMA);
+}
+
+void test_invalid_backup_type(void)
+{
+	ini_config = (config_t *)malloc(sizeof(config_t));
+	ini_config->postgres_config = (postgres_t *)malloc(sizeof(postgres_t));
+	ini_config->smtp_config = (smtp_t *)malloc(sizeof(smtp_t));
+	ini_parse_string("[postgres]\nbackup_type=wrong_type", handler,
+			 ini_config);
+	CU_ASSERT_NOT_EQUAL(ini_config->postgres_config->backup_type, SCHEMA);
+	CU_ASSERT_NOT_EQUAL(ini_config->postgres_config->backup_type, FULL);
+}
```

Comment:
Not sure, but we  can now assert if the program returns the line of the invalid config, right?

Reply:
I can remove these 2 asserts, and keep just the resulting line.

Reply:
Yes, please do.